internal/aggregated-co/builder.go

```diff
@@ -0,0 +1,222 @@
+package aggregated_co
+
+import (
+	"reflect"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// From https://github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorstatus/builder.go
+// Note: the clock api-machinery library in that package is now deprecated, so it has been removed here.
+
+// NewBuilder returns a builder for ClusterOperatorStatus.
+func NewBuilder() *Builder {
+	return &Builder{}
+}
+
+// Builder helps build ClusterOperatorStatus with appropriate
+// ClusterOperatorStatusCondition and OperandVersion.
+type Builder struct {
+	status *configv1.ClusterOperatorStatus
+}
+
+// GetStatus returns the ClusterOperatorStatus built.
+func (b *Builder) GetStatus() *configv1.ClusterOperatorStatus {
+	return b.status
+}
+
+// WithProgressing sets an OperatorProgressing type condition.
+func (b *Builder) WithProgressing(status configv1.ConditionStatus, message string) *Builder {
+	b.init()
+	condition := &configv1.ClusterOperatorStatusCondition{
+		Type:               configv1.OperatorProgressing,
+		Status:             status,
+		Message:            message,
+		LastTransitionTime: metav1.NewTime(time.Now()),
```

Comment:
`LastTransitionTime` should only change when the `Status` for this `Type` changes. When we build the status during a reconcile, do we account for the object's existing status conditions to ensure we leave `LastTransitionTime` unmodified when the status doesn't change?

Reply:
Yes, we should definitely do that